game/world/objects/interactive.cpp

```diff
@@ -215,6 +215,24 @@ void Interactive::tick(uint64_t dt) {
     return;
   if(p->user==nullptr && (state==stateNum && p->attachMode))
     return;
+
+  if (isLadder()) {
+    if (ladderFreeze && state>-1 && state<stateNum+1)
+      return;
+    if (state==stateNum && !reverseState) {
+      auto&  pl=*p->user;
+      auto  pos = pl.position();
+      float rot = pl.rotationRad();
+      float s   = std::sin(rot), c = std::cos(rot);
+      Tempest::Vec3 dp(s,0.0f,-c);
+      pos += dp*6.f*float(dt);
```

Comment:
`6.f` is arbitrary, right? While it's achieving the purpose, I have droughts about correctness. 
On previous iteration, (back at `nextState` days), I found that position of npc changes after detaching from ladder. We should probably exam closely animations here.

Reply:
Currently it's identical to marvin k. I tested some ladders in the G1 swamp camp and one only worked with 6f.


Hmm i have no other idea than "cheating" the character upwards. I only remember that devs said ladders were a nightmare so they left them out of G2 :D.  My test ladder in vanilla G1 funnily didn't work too and dropped the character back down at the top. 

I look into some other ladders in vanilla to see whats happening there